### ISSUE TITLE
🐞 Draw shadow under window

### ISSUE THREAD

**Issue by jrc03c** [vemonet/EmojiMart#8]
### Describe the bug

I'm not actually sure if this is a bug that needs to be reported or a feature that needs to be requested, but would it be possible to show shadows under the window? I'm using Linux Mint 21.2 w/ Cinnamon, and all other windows draw shadows _except_ Emoji Mart. When Emoji Mart launches over top of another window that has the same background color, it's very hard to differentiate the Emoji Mart window from the other window. Here's a screenshot of what I mean:

![](https://i.ibb.co/ZxS9NVv/emoji-mart-screenshot.png)

Thank you!

### Operating System

Linux Mint 21.2 Cinnamon

### Emoji Mart version

0.2.4

**Comment by vemonet** [comment]
Hi @jrc03c thanks for the notification, it's odd that only the EmojiMart window has no shadow

I checked and the lib to draw shadow around tauri app windows does not support Linux: https://github.com/tauri-apps/window-shadows

They say you "would need to go through your composer", so I guess either there is a change to make in the Cinnamon composer config, or a bug where the Cinnamon composer does not draw shadow properly for Tauri apps

I am using Gnome personally, so no idea how to configure the Cinnamon composer

**Comment by jrc03c** [comment]
Ah, okay. Well, thanks for looking into it! I don't know how to use the Cinnamon compositor either. It's interesting that it says "shadows are controlled by the compositor installed on the end-user system," but that seems not to be the case since Cinnamon draws shadows underneath all other windows but _not_ Emoji Mart. That makes me think that either (1) it truly is a Tauri issue, or (2) maybe Cinnamon doesn't draw shadows for windows that lack a frame or something? I'm really not sure. Anyway, thanks for checking it out! Feel free to close this issue if you like. :)

**Comment by jrc03c** [comment]
Hm. Actually, I'm testing it out on Gnome right now, and it's not displaying a shadow here either...

![emoji-mart-gnome](https://github.com/vemonet/EmojiMart/assets/255934/e502a4c9-4c97-459e-8023-ab7dab0842c9)

**Comment by vemonet** [comment]
Interesting, I always use it in dark mode and never noticed this lack of shadow, it must be more impactful on light mode

Maybe a quick fix could be to add a fine borderline, I'll check next time I open the project

The most probable is that it is a bug between Tauri (the framework to build the UI) and the composer. And there your best bet to get it fixed will be to post an issue in the main repo: https://github.com/tauri-apps/tauri/issues/


Otherwise the big question is: do you use **x11 or Wayland?** If you don't know then you probably use wayland

There is a rule in the Wayland world: any display issue always comes from the composer!

It's pretty much all the time the answer you will see when you try to do something on Linux desktop UIs nowadays, because obviously it's much easier to send the fault on someone else than actually fixing it in our own systems ;) 

From my experience building this software, the display libs landscape on Linux is completely broken right now, and that is really sad... Since the move from x11 to wayland for "security reason" we have been just going backward sadly. 

This really don't motivate to work on more desktop stuff... Because the stack we are building on is just poorly built using the usual bad excuse of "it's for your security", but I came to Linux for freedom, power and fame, not security!!

**Comment by jrc03c** [comment]
Hey! I just took a look at the [tauri-apps/window-shadows](https://github.com/tauri-apps/window-shadows) library, and they mention that:

> As of tauri@v2 and recent versions of tao and winit, they all support enablind/disabling shadows so this crate is not needed.

Have you tried upgrading your app to Tauri v2 to see if that fixes the shadow problem? Thanks!

**Comment by vemonet** [comment]
@jrc03c yes the `main` branch is updated to the latest v2 release candidate :) and the flatpak repo `dev` branch is also updated and working for tauri v2 (only tested on x11, not on wayland for now)

I did not publish it because during the v2 beta there was a few issues with the clipboard, and I would like it to be stable and decently tested before doing a new release

So I am waiting for the first official release to publish a new version (they are at the release candidate phase right now so that should be quite soon), but you can already build from source
